### PR TITLE
RandomCsvRowStep keep on choosing the first row

### PR DESCRIPTION
**Pull request by franz1981**
RandomCsvRowStep.Builder is stucked parsing the same first line again and again.

Added tests to verify different `invoke`s select different lines (and these are correct) and that parsing is correctly loading all lines.

Spotted another bug on parsing that doesn't correctly load quoted values spanning multiple lines

### CODE REVIEW

File: core/src/main/java/io/hyperfoil/core/generators/RandomCsvRowStep.java [hunk 1]
```diff
@@ -31,10 +32,17 @@
...
 
-   public RandomCsvRowStep(String[][] rows, ObjectAccess[] columnVars) {
+   public RandomCsvRowStep(String[][] rows, ObjectAccess[] columnVars, IntUnaryOperator rowSelector) {
       this.rows = rows;
       this.columnVars = columnVars;
+      this.rowSelector = rowSelector;
+   }
+
+   // Visible for testing
+   public String[][] rows() {
```
Comment (jesperpedersen) [inline core/src/main/java/io/hyperfoil/core/generators/RandomCsvRowStep.java:44]: Can this be package protected instead ?

File: core/src/main/java/io/hyperfoil/core/generators/RandomCsvRowStep.java [hunk 2]
```diff
@@ -142,12 +155,18 @@ public List<Step> build() {
...
             ObjectAccess[] columnVars = builderColumns.stream().filter(Objects::nonNull).map(SessionFactory::objectAccess).toArray(ObjectAccess[]::new);
-            return Collections.singletonList(new RandomCsvRowStep(rows, columnVars));
+            return Collections.singletonList(new RandomCsvRowStep(rows, columnVars, rowSelector));
          } catch (IOException ioe) {
             throw new BenchmarkDefinitionException("Failed to read file " + file, ioe);
          }
       }
 
+      // Visible For Testing
+      public Builder customSelector(IntUnaryOperator selector) {
```
Comment (jesperpedersen) [inline core/src/main/java/io/hyperfoil/core/generators/RandomCsvRowStep.java:165]: Package protected ?

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by franz1981** [comment]
@jesperpedersen thanks for reviewing! added an additional commit to address comments

**Comment by jesperpedersen** [comment]
Please, squash and force push

**Comment by jimma** [comment]
Only some minor checkstyle issue : 
```[INFO] Starting audit...
/code/Hyperfoil/core/src/test/java/io/hyperfoil/core/generators/RandomCsvRowStepTest.java:51: 'VARIABLE_DEF' should be separated from previous statement.
Audit done.```

**Comment by franz1981** [comment]
@jesperpedersen it seems serialization must be fixed due the changes. I will add a test as well, so please don't merge it yet!

**Comment by franz1981** [comment]
@jesperpedersen waiting user approval and this is done :+1:

**Comment by jesperpedersen** [comment]
checkstyle...

**Comment by jesperpedersen** [comment]
Still failing -- are you using `mvn clean package` ?

**Comment by franz1981** [comment]
@franz1981 yep, just needed last push

**Comment by franz1981** [comment]
```
io.hyperfoil.http.steps.HttpRequestTest
```
this is failing but seems unrelated, I can take a look

**Comment by jesperpedersen** [comment]
Its a transitive failure... but it needs investigation.

**Comment by franz1981** [comment]
But is intermittent in the CI, or...always happen?

**Comment by jesperpedersen** [comment]
Rebased, and merged.

Thanks for your contribution !

**Pull request closed without merging**